Project in personal namespace fails. Fix

Stanislav2017 opened the pull request:
Signed-off-by: Stanislav2017 <stanislavchichek@gmail.com>

<!--
Thank you for helping to improve Crossplane!

Please read through https://git.io/fj2m9 if this is your first time opening a
Crossplane pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes

I this PR. I changed register first char of the method name to lower case for "lateInitialize" and "isProjectUpToDate" and moved the methods to the controller as it is a part of it and is used only in it.
Removed func "updateExternalName" and start use standarв crossplane capability for set external name. I used this path of code

kubernet tries to set the value and if the types do not match it returns an error. But if omitempty is added it sets default value for each type

<!--StartFragment-->
meta.SetExternalName(cr, strconv.Itoa(prj.ID))
return managed.ExternalCreation{ExternalNameAssigned: true}, errors.Wrap(err, errKubeUpdateFailed)
<!--EndFragment-->

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open Crossplane issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->
Fixes https://github.com/crossplane-contrib/provider-gitlab/issues/10

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

manually

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->

[contribution process]: https://git.io/fj2m9

janwillies commented:
Thanks for the PR @Stanislav2017! Looks very good 👍 Could you please update the description and provide a small summary what you did in this PR? e.g. how did you solve the issue (presumably by adding `omitempty`)? I also see that you moved some functions to the controller and simplified `setExternalName`

Stanislav2017 commented:
> Thanks for the PR @Stanislav2017! Looks very good 👍 Could you please update the description and provide a small summary what you did in this PR? e.g. how did you solve the issue (presumably by adding `omitempty`)? I also see that you moved some functions to the controller and simplified `setExternalName`

ok. I added short description

janwillies commented:
This looks very good! Thanks a lot for fixing the issue
